backend/multimodalexplorer/functions/reduce_embed_dims.py

```diff
@@ -41,44 +43,67 @@ def __init__(
         self.embed_file = embed_file
         self.umap_file = umap_file
         self.umap_args = umap_args
+        self.kmean_args = kmean_args
 
-    def load_embeddings(self) -> Optional[np.ndarray]:
+    def _cluster_embed(self, embeddings):
         """
-        Load UMAP embeddings from the stored file.
+        Cluster embeddings using K-Means.
+
+        Args:
+            embeddings (numpy.ndarray): Input embeddings.
 
         Returns:
-            np.ndarray or None: The loaded UMAP embeddings or None if an error occurred.
+            numpy.ndarray: Embeddings with additional cluster labels.
         """
-        dir_path, ext = self.umap_file.values()
-        file_path = get_file_path(dir_path, ext)
+        # Create K-Means model
+        n_clusters, n_init, random_state = self.kmean_args.values()
+        kmeans = KMeans(n_clusters=n_clusters, n_init=n_init, random_state=random_state)
 
-        try:
-            with open(file_path, "rb") as file:
-                embeddings = np.load(file)
-            logger.info(f"Loaded UMAP embeddings with shape: {embeddings.shape}")
-            return embeddings
-        except FileNotFoundError:
-            logger.error(f"UMAP embeddings file not found at: {file_path}")
-            return None
-        except Exception as e:
-            logger.exception(f"An error occurred while loading UMAP embeddings: {e}")
-            return None
+        # Perform clustering
+        clusters = kmeans.fit_predict(embeddings)
```

Comment:
https://umap-learn.readthedocs.io/en/latest/clustering.html suggests that density clustering is better suited for umapped data. We might get some better results with HDBSCAN as per their example code.